engine/src/lib.rs

```diff
@@ -547,33 +688,43 @@ mod contract {
                 &mut Runtime,
             );
         }
+
+        update_hashchain(&mut io, function_name!(), &input, &[], &Bloom::default());
```

Comment:
We need to be careful about the outputs. The output is not empty in the case of this function. You can see the `io` object is used [inside receive_erc20_tokens](https://github.com/aurora-is-near/aurora-engine/blob/develop/engine/src/engine.rs#L781). I think I mentioned this before, but I still wonder if it makes sense to try and connect the hashchain with the `IO` implementation of `NearRuntime` so that we never miss outputs.

I think the issue we ran into before is how to pass in the additional information (input, function name, bloom) without breaking the `IO` API. I'm wondering if there is a way to do a sort of builder pattern so that each call to the `NearRuntime` object updates the piece of the hashchain it knows about, but maybe that will be awkward and not work out well either.

In any case, if we are manually passing outputs then we will need to be very careful that all outputs are properly captured.

Reply:
As per our conversation, capturing the output here and in others would require an investigation of how they look on the standalone side (`get_output_and_log_bloom` method).